controllers/osbuildconfig_controller_test.go

```diff
@@ -0,0 +1,575 @@
+
+	AfterEach(func() {
+		osbuildConfigInstance.DeletionTimestamp = nil
+		osbuildConfigInstance.Status = osbuildv1alpha1.OSBuildConfigStatus{}
+		osbuildInstance.Status = osbuildv1alpha1.OSBuildStatus{}
+	})
+
+	Context("Failure to get OSBuildConfig", func() {
+		It("Should return Done when the instance is not found", func() {
+			// given
+			osBuildConfigRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(nil, errNotFound)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultDone))
+
+		})
+
+		It("Should return requeue when failed to get the instance", func() {
+			// given
+			osBuildConfigRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(nil, errFailed)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultShortRequeue))
+		})
+	})
+
+	Context("Handle deletion", func() {
+		It("Should return Done if the OSBuild CR was deleted", func() {
+			// given
+			osbuildConfigInstance.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+			osBuildConfigRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(osbuildConfigInstance, nil)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultDone))
+		})
+	})
+
+	Context("New OSBuild instance is needed", func() {
+		DescribeTable("should be requeue when failing to patch update", func(lastKnownUserConfiguration *osbuildv1alpha1.UserConfiguration, annotation map[string]string) {
+			// given
+			// TODO fix reflect compare
+			osbuildConfigInstance.Status.LastKnownUserConfiguration = lastKnownUserConfiguration
+			osbuildConfigInstance.Spec.Details.Customizations = &osbuildv1alpha1.Customizations{Packages: []string{"pkg1", "pkg2"}}
+			osbuildConfigInstance.Annotations = annotation
+			osBuildConfigRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(osbuildConfigInstance, nil)
+			osBuildConfigRepository.EXPECT().PatchStatus(requestContext, osbuildConfigInstance, gomock.Any()).Return(errFailed)
+			// when
+			result, err := reconciler.Reconcile(requestContext, request)
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(resultShortRequeue))
+		},
+			Entry("because of LastKnownUserConfiguration is nil", nil, map[string]string{}),
+			Entry("because of LastKnownUserConfiguration is different from the current userConfiguration", &osbuildv1alpha1.UserConfiguration{Customizations: &osbuildv1alpha1.Customizations{Packages: []string{"pkg1"}}}, map[string]string{}),
+			Entry("because of the webHookAnnotationKey was changed", &osbuildv1alpha1.UserConfiguration{Customizations: &osbuildv1alpha1.Customizations{Packages: []string{"pkg1", "pkg2"}}}, map[string]string{"last_webhook_trigger_ts": "1111"}),
```

Comment:
Another case is when both changed in the same Reconcile

Reply:
As I wrote before - the trigger webhook cannot change the object... 
Not sure how it can happen

Reply:
It does, it updates the annotation

Reply:
but the user configuration doesn't contain the annotation only the template and customization

Reply:
When I said both, I meant both `LastKnownUserConfiguration is different` and `webHookAnnotationKey has changed`

Reply:
I updated it